phoenix-core/src/main/java/org/apache/phoenix/iterate/TableResultIterator.java

```diff
@@ -186,20 +188,45 @@ public Tuple next() throws SQLException {
                 if (lastTuple != null) {
                     ImmutableBytesWritable ptr = new ImmutableBytesWritable();
                     lastTuple.getKey(ptr);
+                    if (scan.getStartRow() != null
+                            && Bytes.compareTo(scan.getStartRow(), HConstants.EMPTY_START_ROW) != 0) {
+                        if (scan.includeStartRow()) {
+                            if (Bytes.compareTo(scan.getStartRow(), ptr.get()) > 0) {
+                                throw new QueryOutOfScanRangeException("Query returned is out of scan range");
+                            }
+                        } else {
+                            if (Bytes.compareTo(scan.getStartRow(), ptr.get()) >= 0) {
+                                throw new QueryOutOfScanRangeException("Query returned is out of scan range");
+                            }
+                        }
+                    }
+
+                    if (scan.getStopRow() != null
+                            && Bytes.compareTo(scan.getStopRow(), HConstants.EMPTY_END_ROW) != 0) {
+                        if (scan.includeStopRow()) {
```

Comment:
What is the difference between `scan. getStopRow() != null` and `scan.includeStopRow()` ? 
Can we exclude one check if both are the same?